torch/nn/functional.py

```diff
@@ -2457,6 +2457,11 @@ def interpolate(input, size=None, scale_factor=None, mode='nearest', align_corne
             when :attr:`scale_factor` is kept the same. This only has an effect when :attr:`mode`
             is ``'linear'``, ``'bilinear'``, ``'bicubic'`` or ``'trilinear'``.
             Default: ``False``
+        use_scale_factor (bool, optional): When scale_factor is passed as a parameter, it can be
+            directly used in the output computation, or can be used to compute the output size which
+            will later be used to infer new scales values.
+            If set to ``True``, scale_factors are used in the interpolation.
+            If set to ``False``, the output_size is computed and new scales are infered for the interpolation.
```

Comment:
What happens when `use_scale_factor` is None?

Reply:
The default value is False for now and will be changed to default to True in 1.5.0

Reply:
@zou3519, for more context:
This PR introduces a change in the computation of interpolate, that would had been BC breaking without the added parameter 'use_scale_factor'. 

We had a meeting to discuss this change with @houseroad @ailzhang @fmassa @spandantiwari;
The outcome was to add a parameter (that default to False) so the change is not a breaking change in 1.4.0. 

So 'use_scale_factor' defaults to False in 1.4.0 to keep the old behavior, and we warn the users that the default behavior will change in version 1.5.0.

Reply:
Thanks for the context. If the default is `false` we should record in here that the default is false.

I'm having trouble understanding from the explanation in the documentation what exactly is going to be different -- does the value of the output tensor change depending on `use_scale_factor`?

Reply:
Yes the behavior is different and affects the output.

Interpolate uses scale values to compute the output. 
Interpolate can be called with scale_factor param, which are the scales to use in the interpolation computation.

- with use_scale_factor = False (old behavior that we wish the deprecate):
The output_size is computed as : output_size = input_size * scale_factor
Then scales are computed later in the aten op : scales = output_size / input_size
With this behavior there is a loss in precision in the computed scales.


- with use_scale_factor = True (new desired behavior):
scale_factor is passed to the aten op and scales = scale_factor

The scales used in the interpolation being different, the output of interpolate will differ in certain cases, which is why it is important to explain this param in the doc and warn the user about the behavior change in 1.5.0.

Reply:
Okay, I see. It was not clear to me how the new scales could be different from the input scales (from your explanation they can be different due to floating point precision). That might be a good thing to note in the docs?

But at any rate, I understand the value of the new parameter, thank you for clarifying.

Reply:
Great, thanks. Lmk if there are other suggestions, otherwise I will push a new commit with the doc updated as suggested.

Reply:
I read some more context from https://github.com/pytorch/pytorch/issues/27176. To clarify, is the output different only when doing ONNX export, or is it different when running pytorch python as well?

Reply:
This issue was opened before we discovered the precision loss in the scales.
Now the output is also different in Pytorch when use_scale_factor is True